docs/Compiler_Hardening_Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -239,9 +240,17 @@ Both `_FORTIFY_SOURCE=1` and `_FORTIFY_SOURCE=2` are expected to have a negligib
 
 #### When not to use?
 
-`_FORTIFY_SOURCE` is recommended for all application that depend on glibc.
+`_FORTIFY_SOURCE` is recommended for all application that depend on glibc and should be widely deployed. Most packages in all major Linux distributions enable at least `_FORTIFY_SOURCE=2` and some even enable `_FORTIFY_SOURCE=3`. There are a couple of situations when `_FORTIFY_SOURCE` may break existing applications:
 
-However, when enabling `_FORTIFY_SOURCE=2` in existing code bases regression testing should be used to ensure the run-time checks do not adversely affect existing features.
+- If the fortified glibc function calls show up as hotspots in your application performance profile, there is a chance that `_FORTIFY_SOURCE` may have a negative performance impact. This is not a common or widespread slowdown[^29] but worth keeping in mind if slowdowns are observed due to this option
+- Applications that use the GNU extension for flexible array members in structs[^30] may confuse the compiler into thinking that an object is smaller than it actually is, resulting in spurious aborts. The safe resolution for this is to port these uses to C99 flexible arrays but if that is not possible (e.g. due to the need to support a compiler that does not support C99 flexible arrays), one may need to downgrade or disable `_FORTIFY_SOURCE` protections.
```

Comment:
Is this caveat specific to `_FORTIFY_SOURCE=3`? 

If so, should that be stated explicitly and the statement about downgrading say "downgrade to ´_FORTIFY_SOURCE=2´"?.

Reply:
It's not specific to `_FORTIFY_SOURCE=3`; it's just that applications are more likely to hit these issues with `_FORTIFY_SOURCE=3` because the coverage is significantly higher.  A number of applications disabled fortification before `3` was introduced and so far that list has not increased in my F38 testing.